rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/NestedParticipantIT.java

```diff
@@ -24,78 +24,73 @@
 
 import io.narayana.lra.arquillian.resource.NestedParticipant;
 import io.narayana.lra.arquillian.spi.NarayanaLRARecovery;
-import io.narayana.lra.client.NarayanaLRAClient;
 import org.eclipse.microprofile.lra.annotation.LRAStatus;
 import org.eclipse.microprofile.lra.annotation.ws.rs.LRA;
-import org.eclipse.microprofile.lra.tck.service.LRAMetricService;
 import org.eclipse.microprofile.lra.tck.service.LRAMetricType;
 import org.jboss.arquillian.container.test.api.Deployment;
-import org.jboss.arquillian.junit.Arquillian;
 import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.logging.Logger;
 import org.jboss.shrinkwrap.api.spec.WebArchive;
-import org.junit.After;
 import org.junit.Assert;
-import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
-import org.junit.runner.RunWith;
+import org.junit.rules.TestName;
 
-import javax.inject.Inject;
 import javax.ws.rs.NotFoundException;
 import javax.ws.rs.WebApplicationException;
-import javax.ws.rs.client.Client;
-import javax.ws.rs.client.ClientBuilder;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.UriBuilder;
+import java.io.UnsupportedEncodingException;
 import java.net.URI;
 import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
 
 import static javax.ws.rs.core.Response.Status.NOT_FOUND;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotEquals;
 import static org.junit.Assert.assertNull;
 
-@RunWith(Arquillian.class)
-public class NestedParticipantIT {
+public class NestedParticipantIT extends TestBase {
 
-    @ArquillianResource
-    private URL baseURL;
-
-    @Inject
-    private LRAMetricService lraMetricService;
```

Comment:
We are not going away from CDI in LRA Participants. Removing:
```
@Inject
private LRAMetricService lraMetricService;
```
from this test class is a reflection of the new design I went for: `NestedParticipantIT` is now `RunAsClient` and, therefore, CDI is not available (that's also the reason of the modifications in the LRA participant `NestedParticipant` to include `LRAMetricService` in the REST API). Basically, LRA participants are now hosted in the Arquillian container while test classes are run as client (`@RunAsClient`)

Reply:
However, this is not something that we should go for. We will need CDI for participants because of the non-JAX-RS cases, interceptors, etc. If it isn't possible to use CDI in RunAsClient than we shouldn't use it. cc/ @ochaloup